### PR TITLE
[CodingStyle] Register UnSpreadOperatorRector to coding-style config set

### PR DESCRIPTION
**Pull request by samsonasik**

### CODE REVIEW

**Comment by samsonasik** [comment]
It cause phpstan errors:

```php

 ------------------------------------------------------------------------
  packages/testing/src/Finder/RectorsFinder.php:99
 ------------------------------------------------------------------------
  - '#Parameter \#1 \.\.\.\$paths of method Nette\\Loaders\\RobotLoader\:\:addDirectory\(\) expects string, array<string\> given#'
 ------------------------------------------------------------------------

 ------------------------------------------------------------------------
  src/Autoloading/AdditionalAutoloader.php:107
 ------------------------------------------------------------------------
  - '#Parameter \#1 \.\.\.\$paths of method Nette\\Loaders\\RobotLoader\:\:addDirectory\(\) expects string, array<string\> given#'
 ------------------------------------------------------------------------
 ```
 
 @TomasVotruba do we need to make configurable exclude specific classes ?

**Comment by TomasVotruba** [comment]
This rule should skip cases, where class method is in `/vendor`

**Comment by samsonasik** [comment]
I've updated to exclude vendor directory

**Comment by samsonasik** [comment]
oh, the check need to be in caller (MethodCall) too

**Comment by samsonasik** [comment]
Rectify got the following error:

```bash
[applying] Rector\DeadCode\Rector\Property\RemoveSetterOnlyPropertyAndMethodCallRector
PHP Fatal error:  Uncaught ReflectionException: Class Rector\Core\PhpParser\NodeTraverser\AnonymousClass9afdd4ad99cd6d89e2f8831d301b14c5 does not exist in /home/runner/work/rector/rector/rules/coding-style/src/Rector/ClassMethod/UnSpreadOperatorRector.php:101
Stack trace:
#0 /home/runner/work/rector/rector/rules/coding-style/src/Rector/ClassMethod/UnSpreadOperatorRector.php(101): ReflectionClass->__construct('Rector\\Core\\Php...')
#1 /home/runner/work/rector/rector/rules/coding-style/src/Rector/ClassMethod/UnSpreadOperatorRector.php(166): Rector\CodingStyle\Rector\ClassMethod\UnSpreadOperatorRector->getClassFileNameByClassMethod(Object(PhpParser\Node\Stmt\ClassMethod))
#2 /home/runner/work/rector/rector/rules/coding-style/src/Rector/ClassMethod/UnSpreadOperatorRector.php(137): Rector\CodingStyle\Rector\ClassMethod\UnSpreadOperatorRector->isInVendor(Object(PhpParser\Node\Stmt\ClassMethod))
#3 /home/runner/work/rector/rector/rules/coding-style/src/Rector/ClassMethod/UnSpreadOperatorRector.php(83): Rector\CodingStyle\Rector\Clas in /home/runner/work/rector/rector/rules/coding-style/src/Rector/ClassMethod/UnSpreadOperatorRector.php on line 101
Fatal error: Uncaught ReflectionException: Class Rector\Core\PhpParser\NodeTraverser\AnonymousClass9afdd4ad99cd6d89e2f8831d301b14c5 does not exist in /home/runner/work/rector/rector/rules/coding-style/src/Rector/ClassMethod/UnSpreadOperatorRector.php:101
Stack trace:
#0 /home/runner/work/rector/rector/rules/coding-style/src/Rector/ClassMethod/UnSpreadOperatorRector.php(101): ReflectionClass->__construct('Rector\\Core\\Php...')
#1 /home/runner/work/rector/rector/rules/coding-style/src/Rector/ClassMethod/UnSpreadOperatorRector.php(166): Rector\CodingStyle\Rector\ClassMethod\UnSpreadOperatorRector->getClassFileNameByClassMethod(Object(PhpParser\Node\Stmt\ClassMethod))
#2 /home/runner/work/rector/rector/rules/coding-style/src/Rector/ClassMethod/UnSpreadOperatorRector.php(137): Rector\CodingStyle\Rector\ClassMethod\UnSpreadOperatorRector->isInVendor(Object(PhpParser\Node\Stmt\ClassMethod))
#3 /home/runner/work/rector/rector/rules/coding-style/src/Rector/ClassMethod/UnSpreadOperatorRector.php(83): Rector\CodingStyle\Rector\Clas in /home/runner/work/rector/rector/rules/coding-style/src/Rector/ClassMethod/UnSpreadOperatorRector.php on line 101
```

**Comment by samsonasik** [comment]
fixed.

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

**Review by TomasVotruba — APPROVED** [review]

**Comment by TomasVotruba** [comment]
Thank you! Could you add also test case for this so we have it covered?

**Comment by samsonasik** [comment]
I've added fixture for skip vendor

File: rules/coding-style/tests/Rector/ClassMethod/UnSpreadOperatorRector/Fixture/skip_vendor.php.inc [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+final class SkipVendor
+{
+    public function run(...$args)
+    {
+    }
+
+    public function execute(array $data)
+    {
+        $this->run(...$data);
+    }
```
Comment (TomasVotruba) [inline rules/coding-style/tests/Rector/ClassMethod/UnSpreadOperatorRector/Fixture/skip_vendor.php.inc:14]: In this case, the `...` should have been removed actually. It's the same file


The test I had in mind was using the `RobotLoader->addDirectory(...$directories)`

Comment (samsonasik) [reply rules/coding-style/tests/Rector/ClassMethod/UnSpreadOperatorRector/Fixture/skip_vendor.php.inc]: implemented.

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba I think it is ready.

**Comment by TomasVotruba** [comment]
Thank you :+1:

**Pull request merged**